src/storage/src/source/types.rs

```diff
@@ -157,7 +157,9 @@ impl crate::healthcheck::HealthStatus for HealthStatusUpdate {
         if let Some(other) = other {
             self.update != other.update
         } else {
-            true
+            // Reduces the likelihood of duplicate entries
+            // from misuse.
+            self.update != HealthStatus::Starting
```

Comment:
OTOH, this one seems fine, since `self` is the new status and `other` is the old status IIUC.

Reply:
this is actually the exact same order of logic, but its very confusing looking